enumerable.rb

```diff
@@ -0,0 +1,108 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < self.size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    (self.size - 1).times do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    item = []
+    self.my_each do |i|
+      item << i if yield(i)
+    end
+    item
+  end
+
+  def my_all?
```

Comment:
* [ ] Run your methods against these examples,
```  p %w[ant bear cat].my_all? { |word| word.length >= 3 } #=> true
     p %w[ant bear cat].my_all? { |word| word.length >= 4 } #=> false
     p %w[ant bear cat].my_all?(/t/)                        #=> false
     p [1, 2i, 3.14].my_all?(Numeric)                       #=> true
     p [nil, true, 99].my_all?                              #=> false
     p [].my_all?                                           #=> true
```
compare the result with
```  p %w[ant bear cat].all? { |word| word.length >= 3 } #=> true
     p %w[ant bear cat].all? { |word| word.length >= 4 } #=> false
     p %w[ant bear cat].all?(/t/)                        #=> false
     p [1, 2i, 3.14].all?(Numeric)                       #=> true
     p [nil, true, 99].all?                              #=> false
     p [].all?                                           #=> true```
both ```.all``` and ```.my_all``` should be same